src/games/strategy/triplea/ai/proAI/data/ProMoveOptions.java

```diff
@@ -0,0 +1,116 @@
+package games.strategy.triplea.ai.proAI.data;
+
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.ai.proAI.util.ProBattleUtils;
+import games.strategy.triplea.ai.proAI.util.ProUtils;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.util.Match;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ProMoveOptions {
+
+  private final ProUtils utils;
+  private final ProBattleUtils battleUtils;
+  private final Map<Territory, ProTerritory> maxMoveMap;
+  private final Map<Territory, List<ProTerritory>> moveMaps;
+
+  public ProMoveOptions(final ProUtils utils, final ProBattleUtils battleUtils) {
+    this.utils = utils;
+    this.battleUtils = battleUtils;
+    maxMoveMap = new HashMap<Territory, ProTerritory>();
+    moveMaps = new HashMap<Territory, List<ProTerritory>>();
+  }
+
+  public ProMoveOptions(final ProUtils utils, final ProBattleUtils battleUtils,
+      final List<Map<Territory, ProTerritory>> moveMapList, final PlayerID player, final boolean isAttacker) {
+    this(utils, battleUtils);
+    populateMaxMoveMap(moveMapList, player, isAttacker);
+    populateMoveMaps(moveMapList);
```

Comment:
You can get rid of the side effects of the populate methods by returning the maps directly. Then the populate methods could be static, and probably better called 'create' as well. Then line 35 and 36 can go away, and the assignmens on lines 28 and 29 would be updated to call the create method directly.